test/parallel/test-process-redirect-warnings.js

```diff
@@ -0,0 +1,29 @@
+'use strict';
+
+// Tests the --redirect-warnings command line flag by spawning
+// a new child node process that emits a warning into a temporary
+// warnings file. Once the process completes, the warning file is
+// opened and the contents are validated
+
+const common = require('../common');
+const fs = require('fs');
+const fork = require('child_process').fork;
+const path = require('path');
+const assert = require('assert');
+
+common.refreshTmpDir();
+
+const warnmod = require.resolve(common.fixturesDir + '/warnings.js');
+const warnpath = path.join(common.tmpDir, 'warnings.txt');
+
+fork(warnmod, {execArgv: [`--redirect-warnings=${warnpath}`]})
+  .on('exit', common.mustCall(() => {
+    fs.stat(warnpath, common.mustCall((err, stat) => {
```

Comment:
Is the `stat()` call necessary? Why not just let `fs.readFile()` fail if the file isn't there? Reduces the nesting by one. If I'm not mistaken, the "stat() followed by read()" is kind of an anti-pattern that we should probably discourage because there's nothing (in theory) preventing another process from deleting the file between the `stat()` and the `read()`.

Maybe I'm missing something here, but it seems this could just use `fs.readFileSync()` here and be a one- or two-liner that's probably easier to understand.